include/libtorrent/sha1_hash.hpp

```diff
@@ -161,7 +161,7 @@ namespace aux {
 		}
 		bool operator!=(digest32 const& n) const
 		{
-			return !std::equal(n.m_number, n.m_number + number_size, m_number);
```

Comment:
http://www.learncpp.com/cpp-tutorial/96-overloading-the-comparison-operators/
http://courses.cms.caltech.edu/cs11/material/cpp/donnie/cpp-ops.html 

revert all !(*this == x); ?